utils/user_management.py

```diff
@@ -0,0 +1,158 @@
+'''
+Utility for managing user names and passwords for TabPy.
+'''
+
+from argparse import ArgumentParser
+import logging
+import os
+import random
+import sys
+from tabpy_server.app.util import parse_pwd_file
+from tabpy_server.handlers.util import hash_password
+
+logger = logging.getLogger(__name__)
+
+
+def build_cli_parser():
+    parser = ArgumentParser(
+        description=__doc__,
+        epilog='''
+            For more information about how to configure and
+            use authentication for TabPy read documentation
+            at https://github.com/tableau/TabPy
+            ''',
+        argument_default=None,
+        add_help=True)
+    parser.add_argument(
+        'command',
+        choices=['add', 'update'],
+        help='Command to execute')
+    parser.add_argument(
+        '-u',
+        '--username',
+        help='Username to add to passwords file')
+    parser.add_argument(
+        '-f',
+        '--pwdfile',
+        help='Fully qualified path to passwords file')
+    parser.add_argument(
+        '-p',
+        '--password',
+        help=('Password for the username. If not specified password will '
+              'be generated'))
+    return parser
+
+
+def check_args(args):
+    if (args.username is None) or (args.pwdfile is None):
+        return False
+
+    return True
+
+
+def generate_password(pwd_len=16):

```

Comment:
I checked https://docs.python.org/3/library/secrets.html and it shows how to generate a password in the same way we do in this function

Reply:
Yea I was just thinking you could use `token_safe` unless you really cared to control the contents here. Simplifies code

Reply:
If I read the documentation correctly `token_safe` returns base64 encoded token - there won't be punctuation characters in it.

Reply:
There is URL-safe punctuation. You can see the PEP for the module here:

https://www.python.org/dev/peps/pep-0506/#id42

Basically the whole point of the module is to prevent people from using `random` to generate passwords, which is what we are doing here. Random was just kept for backwards compatibility

Reply:
Got it (I think), pushing updated code.